pkg/db/advisory_locks.go

```diff
@@ -0,0 +1,250 @@
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/openshift-online/rh-trex/pkg/logger"
+	"gorm.io/gorm"
+)
+
+type (
+	advisoryLockMap map[string]*AdvisoryLock
+	LockType        string
+)
+
+const (
+	Migrations LockType = "migrations"
+	Dinosaurs  LockType = "dinosaurs"
+	Events     LockType = "events"
+)
+
+type LockFactory interface {
+	NewAdvisoryLock(ctx context.Context, id string, lockType LockType) (string, error)
+	NewUnblockingLock(ctx context.Context, id string, lockType LockType) (string, bool, error)
```

Comment:
I think _nonblocking_ is technically more correct than unblocking.

Reply:
change the name to nonblocking